apps/gitness/src/pages/profile-settings/profile-settings-keys-page.tsx

```diff
@@ -17,15 +20,19 @@ interface SandboxSettingsAccountKeysPageProps {
   openSshKeyDialog: () => void
   openAlertDeleteDialog: (data: AlertDeleteParams) => void
   error: { type: string; message: string } | null
+  headers?: Headers
 }
 const SandboxSettingsAccountKeysPage: React.FC<SandboxSettingsAccountKeysPageProps> = ({
   publicKeys,
   tokens,
   openTokenDialog,
   openSshKeyDialog,
   openAlertDeleteDialog,
-  error
+  error,
+  headers
 }) => {
+  const [page, setPage] = useQueryState('page', parseAsInteger.withDefault(1))
+  const totalPages = parseInt(headers?.get(PageResponseHeader.xTotalPages) || '')
```

Comment:
ideally we should have no state management on a rendering page. We should do this in the `container` and just pass props/callbacks.

Reply:
`container` can have multiple api calls in it and each of those api calls will have associated response headers. Keeping this in the relevant rendering page would just keep it cleaner.

Reply:
Sounds good